DuckDuckGo/PrivacyProtectionController.swift

```diff
@@ -63,6 +67,33 @@ class PrivacyProtectionController: UIViewController {
         embeddedController.pushViewController(controller, animated: true)
     }
 
+    private func showBlockerListError() {
+        guard let controller = storyboard?.instantiateViewController(withIdentifier: "Error") as? PrivacyProtectionErrorController else { return }
+        controller.errorText = UserText.privacyProtectionReloadBlockerLists
+        embeddedController.pushViewController(controller, animated: true)
+        showLoadBlockerListTryAgain(controller)
+    }
+
+    private func showLoadBlockerListTryAgain(_ controller: PrivacyProtectionErrorController) {
+        controller.onTryAgain {
+            let blockerListLoader = BlockerListsLoader()
+            blockerListLoader.start { newData in
+                self.handleLoadBlockerListResult(controller, blockerListLoader.hasData)
```

Comment:
The strong reference to self will likely cause a memory leak. I ran the profiler and the privacy protection dashboard is never cleared from memory. There are a few strong references to self that probably need a review to make sure they are safe.
<img width="508" alt="screen shot 2017-12-15 at 12 37 23" src="https://user-images.githubusercontent.com/1652407/34043063-4ab815da-e197-11e7-9797-1a7ffd768c89.png">

Reply:
Ah, amazing. Thanks @subsymbolic

Reply:
I don't think this will leak as the closure is not retained, but the user might dismiss the view controller while the update is progress which could cause a crash.